src/lib/auth.ts

```diff
@@ -255,10 +255,27 @@ export const authenticate = (appMode: AppMode, appConfig: AppConfigModelType, ur
       if (!fakeClass || !fakeUser) {
         return reject("Missing fakeClass or fakeUser parameter for demo!");
       }
-      const [userType, userId] = fakeUser.split(":");
+      const userType = fakeUser.split(":")[0];
+      let userId = fakeUser.split(":")[1];
+
       if (((userType !== "student") && (userType !== "teacher")) || !userId) {
         return reject("fakeUser must be in the form of student:<id> or teacher:<id>");
       }
+
+      if ((userId === "random")) {
+        const url = window.location.toString();
+        const title = document.title;
+        const randomStudentId = uniqueId();
```

Comment:
I'm not quite sure why, but 16 character user ids seems like a lot to me. I was thinking 8 would be sufficient for demo users and would make the urls seem a bit less imposing.

Reply:
Then let's stick with 16. It's not worth shaving a couple characters.